geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/api/package.scala

```diff
@@ -324,6 +324,7 @@ package object api {
   // specialize long to avoid boxing for z2/xz2 index
   case class BoundedRange[@specialized(Long) T](lower: T, upper: T) extends ScanRange[T]
   case class SingleRowRange[T](row: T) extends ScanRange[T]
+  case class S3BoundedRange[T](lower: T, upper: T) extends ScanRange[T]
```

Comment:
do we need this extra class? seems like we could just use the `BoundedRange` instead

Reply:
yes, now it is extra. we adjust it during coding.